app/src/main/java/de/xikolo/controllers/course/CourseActivity.kt

```diff
@@ -168,6 +173,10 @@ class CourseActivity : ViewModelActivity<CourseViewModel>(), UnenrollDialog.List
         handleCourseDeepLinkTab(intent)
 
         updateViewPagerTab()
+
+        if (Feature.shortcuts()) {
+            ShortcutHelper().addCourse(applicationContext, course.id, course.title)
```

Comment:
Hmm shouldn't this be called in the ViewModels onFirstCreate? @bengelhaupt what do you think?

Reply:
@bengelhaupt can you give your feedback on that? :)

Reply:
It can't be called in `onFirstCreate`, because the call requires a the course title which is only available after the course has been fetched. Furthermore, `setupCourse()` is only called when the course is actually valid (deep link handling), which would not be the case with the viewmodel.

Reply:
Another question: Should the shortcuts contain all visited courses or only the ones where the user is enrolled?

Reply:
Only the enrolled ones, to "continue learning".